ultravox/data/datasets.py

```diff
@@ -159,6 +159,11 @@ def __post_init__(self):
         if self.audio is not None:
             if self.audio.dtype == np.float64:
                 self.audio = self.audio.astype(np.float32)
+            elif self.audio.dtype == np.int16:
+                # Gradio mp3 file submission, as one example, seems to turn an
+                # mp3 into int16, thus it can be useful to convert to float32 if
+                # we come across this.
+                self.audio = self.audio.astype(np.float32) / np.float32(32768.0)
```

Comment:
Seems like the issue is not limited to `int16`. We can get `int32` as well:
```suggestion
                self.audio = self.audio.astype(np.float32) / np.float32(32768.0)
            elif self.audio.dtype == np.int32:
                self.audio = self.audio.astype(np.float32) / np.float32(2147483648)

```

Reply:
Of course another solution is to change Gradio input side:
`gr.Audio(label="Audio", show_download_button=True)` -> add `type="filepath` and load the audio file ourselves.

Reply:
> Of course another solution is to change Gradio input side: `gr.Audio(label="Audio", show_download_button=True)` -> add `type="filepath` and load the audio file ourselves.

There seem to be enough code paths that can lead into this that it's worth adding your suggestion. It may also be worth the Gradio change but if so leaving for another time.